src/compaction.c

```diff
@@ -499,6 +507,56 @@ int RollingMedReadContext(void *contextPtr, RedisModuleIO *io, __unused int encv
     return TSDB_ERROR;
 }
 
+void *RollingAvgCreateContext(__attribute__((unused)) bool reverse, size_t windowSize) {
+    RollingAvgContext *context = (RollingAvgContext *)malloc(sizeof(RollingMedContext));
+    context->count = 0;
+    context->windowSize = windowSize;
+    context->items_queue = carray_new(double, windowSize);
+    context->sum = 0.0;
+
+    for (int i = 0; i < windowSize; i++){
+        carray_push_back(context->items_queue, 0.0);
+    }
+    return context;
+}
+
+void RollingAvgAddValue(void *contextPtr, double value, __attribute__((unused)) timestamp_t ts) {
+    RollingAvgContext *context = (RollingAvgContext *)contextPtr;
+    double* del_ptr = ((double *)context->items_queue->buf) + context->count;
+
+    // remove the oldest value and add the newest to the sum
+    context->sum += value - *del_ptr;
+    context->count = context->count + 1;
+    if(context->count >= context->windowSize)
+        context->count = 0;
+
+    *del_ptr = value;
```

Comment:
I'm not sure I understand this, I suggest using the carray api:
carray_push_back
carray_pop_front